Use 'tree' style stat names, and riak_core_stat_q

russelldb opened the pull request:
Use riak core stat query for riak_kv_stat get_stats
Add new read repair dimensioned stat aggregations to stat blob

Depends on riak_core pr - https://github.com/basho/riak_core/pull/252

russelldb committed:
Use riak core stat query for riak_kv_stat get_stats

Move backwards compatiblity code to own module for easier
deprecation and removal.

russelldb committed:
Add new read repair dimensioned stat aggregations to stat blob

russelldb committed:
Add pipe stats to THE BLOB

beerriot commented on src/riak_kv_stat_bc.erl:
```diff
@@ -0,0 +1,384 @@
+    Key = [element(N, Name) || N <- Fields],
+    join(Key).
+
+recursive_join(BaseName, Aggregates) ->
```
The name `recursive_join` was a little vague for me. Maybe a comment explaining how it's flattening the nested stat tree, creating a flat list of stats with names prepended by `BaseName_`?

beerriot commented on src/riak_kv_stat_bc.erl:
```diff
@@ -0,0 +1,384 @@
+                     end,
+                     [],
+                     Aggregates),
+    lists:flatten(L).
```
It seems like it shouldn't be necessary to call `lists:flatten/1` on every recursion of this function … it should only be needed once at the top level (in `aggregate/3`?).

beerriot commented on src/riak_kv_stat.erl:
```diff
@@ -443,71 +263,16 @@ register_stat(Name, histogram) ->
     folsom_metrics:new_histogram(Name, SampleType, SampleArgs).
 
+%% @doc the histogram sample type may be set in app.config
+%% use key `stat_sample_type` in the `riak_kv` section. Or the
```
These are markdown-style literal escapes, not edoc-style.

russelldb committed:
Use binary bucket names, rather than converting to atoms

Technically there is no bound to the number of buckets one can
create, and no limit the length of a bucket name. The erlang atom
table is a finite resource and atom sapce is not reclaimed so
creating unbounded atoms of any length is a bad idea.

russelldb committed:
Address review comments

Re-name and better comment confusing `recursive_join` fun.
Remove redundant call to `lists:flatten` as once is enough.
Correct markdown style escapes to edoc style.

russelldb commented:
Pushed a commit that addresses review comments.

beerriot commented:
Looks great, +1!
